### PR TITLE
Add option to ignore content-length with multipart body

### PR DESCRIPTION
**Pull request by outamaa**
The current implementation of `multipart::form` requires that the `Content-Length` header exists. However, there are valid situations where this is not the case, for example when `Transfer-Encoding` header is present. See e.g. [RFC2616, section 4.4](https://datatracker.ietf.org/doc/html/rfc2616#section-4.4).

This PR adds an option to not check the content length, thus allowing the `multipart::form` filter to be used in cases where `Content-Length` header is not present. The change does not break the existing API.

### CODE REVIEW

**Comment by jxs** [comment]
Hi, thanks for your interest! and sorry for the delay. I think this should rather be part of https://github.com/seanmonstar/warp/pull/846/ what do you think?

**Comment by outamaa** [comment]
> Hi, thanks for your interest! and sorry for the delay.

No problem. Hope this proves useful to somebody else besides me. :)

> I think this should rather be part of #846 what do you think?

I really have no strong preference one way or the other, though I do think of this as a separate feature to multipart being streamable. How would you suggest going forward with this?

**Comment by xd009642** [comment]
Small bump as I was wondering what's taken to progress this as #846 is closed and multipart stuff in warp seems to have been redone recently but this feature still isn't present :eyes:

**Comment by outamaa** [comment]
I now merged the changes from upstream master into this branch.

File: src/filters/multipart.rs [hunk 1]
```diff
@@ -61,7 +61,13 @@ impl FormOptions {
     ///
     /// Defaults to 2MB.
     pub fn max_length(mut self, max: u64) -> Self {
-        self.max_length = max;
+        self.max_length = Some(max);
+        self
+    }
+
+    /// Set no limit to maximum byte length for the body.
+    pub fn no_max_length(mut self) -> Self {
```
Comment (seanmonstar) [inline src/filters/multipart.rs:69]: We could change to `max_length(impl Into<Option<u64>>)`. All previous calls will still work, but you can now pass `None`. And then document that.

Comment (outamaa) [reply src/filters/multipart.rs:69]: Sounds good, thanks!

Comment (outamaa) [reply src/filters/multipart.rs]: Done.

**Review by seanmonstar — APPROVED** [review]

**Pull request merged**